### ISSUE TITLE
feat: Implement VGG16 base model with ImageNet pre-trained feature extractor

### ISSUE THREAD

**Issue by Bosaj** [Bosaj/Transfer-Learning-VGG16-Dog-Cat-Classification#1]
Freeze convolutional backbone and attach custom dense classification head with dropout regularization.
